analyzers/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs

```diff
@@ -48,63 +45,44 @@ public abstract class MutableFieldsShouldNotBe : SonarDiagnosticAnalyzer
 
-            var assignmentsImmutability = GetFieldAssignmentImmutability(typeDeclaration, fieldDeclarations,
-                analysisContext.SemanticModel);
+            var assignmentsImmutability = FieldAssignmentImmutability(typeDeclaration, fieldDeclarations, analysisContext.SemanticModel);
 
             foreach (var fieldDeclaration in fieldDeclarations)
             {
-                if (!HasAllInvalidModifiers(fieldDeclaration) ||
-                    fieldDeclaration.Declaration.Variables.Count == 0)
-                {
-                    return;
-                }
-
-                if (!(analysisContext.SemanticModel.GetDeclaredSymbol(fieldDeclaration.Declaration.Variables[0]) is IFieldSymbol fieldSymbol) ||
-                    fieldSymbol.Type == null ||
-                    fieldSymbol.GetEffectiveAccessibility() != Accessibility.Public ||
-                    IsImmutableOrValidMutableType(fieldSymbol.Type))
+                if (HasAllInvalidModifiers(fieldDeclaration)
+                    && fieldDeclaration.Declaration.Variables.Count > 0
+                    && analysisContext.SemanticModel.GetDeclaredSymbol(fieldDeclaration.Declaration.Variables[0]) is IFieldSymbol { Type: not null } fieldSymbol
+                    && fieldSymbol.GetEffectiveAccessibility() == Accessibility.Public
+                    && !IsImmutableOrValidMutableType(fieldSymbol.Type)
+                    // The field seems to be violating the rule but we should exclude the cases where the field is read-only
+                    // and all initializations to this field are immutable
+                    && CollectInvalidFieldVariables(fieldDeclaration, assignmentsImmutability, analysisContext.SemanticModel).ToList() is {Count: > 0} incorrectFieldVariables)
                 {
-                    return;
-                }
-
-                // The field seems to be violating the rule but we should exclude the cases where the field is read-only
-                // and all initializations to this field are immutable
-                var incorrectFieldVariables = CollectInvalidFieldVariables(fieldDeclaration, assignmentsImmutability,
-                        analysisContext.SemanticModel)
-                    .ToSentence(quoteWords: true);
-
-                if (incorrectFieldVariables != null)
-                {
-                    analysisContext.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0],
-                        fieldDeclaration.Declaration.Type.GetLocation(), incorrectFieldVariables));
+                    var pluralize = incorrectFieldVariables.Count > 1 ? "s" : string.Empty;
+                    analysisContext.ReportDiagnosticWhenActive(
+                        Diagnostic.Create(SupportedDiagnostics[0],
+                        fieldDeclaration.Declaration.Type.GetLocation(),
+                        pluralize, incorrectFieldVariables.ToSentence(quoteWords: true)));
```

Comment:
should be indented, one arg per line
```suggestion
                    analysisContext.ReportDiagnosticWhenActive(
                        Diagnostic.Create(SupportedDiagnostics[0],
                                                     fieldDeclaration.Declaration.Type.GetLocation(),
                                                     pluralize, 
                                                     incorrectFieldVariables.ToSentence(quoteWords: true)));
```